test/rcd_test/ext/mri/extconf.rb

```diff
@@ -23,6 +23,73 @@
+      ##
+      ## There are good reasons to do this, including faster initialiation/loading as the Darwin
+      ## toolchain gets improved over time.
+      ##
+      ## Unfortunately, this flag prevents us from building a shared object that works with both a
+      ## Ruby compiled with `--enable-shared` and one compiled with `--disabled-shared`. The result
+      ## is usually an "Symbol not found" error about `_rb_cObject`, or a "dyld: missing symbol
+      ## called" error.
+      ##
+      ## There are two workarounds that I know of (there may be others I don't know of), and
+      ## they are ...
+
+      ## ----------------------------------------
+      ## SOLUTION 1, the `-flat_namespace` flag
+      ##
+      ## > Two-level namespace
+      ## >   By default all references resolved to a dynamic library record the library to which
+      ## >   they were resolved. At runtime, dyld uses that information to directly resolve symbols.
+      ## >   The alternative is to use the -flat_namespace option.  With flat namespace, the library
+      ## >   is not recorded.  At runtime, dyld will search each dynamic library in load order when
+      ## >   resolving symbols. This is slower, but more like how other operating systems resolve
+      ## >   symbols.  Two-level namespace By default all references resolved to a dynamic library
+      ## >   record the library to which they were resolved. At runtime, dyld uses that information
+      ## >   to directly resolve symbols.  The alternative is to use the -flat_namespace option.
+      ## >   With flat namespace, the library is not recorded.  At runtime, dyld will search each
+      ## >   dynamic library in load order when resolving symbols. This is slower, but more like how
+      ## >   other operating systems resolve symbols.
+      ##
+      #
+      # puts "Adding '-flat_namespace'"
+      # $LDFLAGS << ' -flat_namespace'
+      #
+      ## This solution unfortunately introduces new behavior that any symbols statically linked into
+      ## the shared object (e.g., libxml2 in nokogiri.bundle) may not be resolved locally from the
+      ## shared object, but instead resolved from a shared object loaded in the main process.
+      ##
+      ## This solution might be good for you if:
+      ## - you don't statically link things into your bundle,
+      ## - or you don't export those symbols,
+      ## - or you can avoid exporting those symbols (e.g., by using `-load_hidden`, or
+      ##   `-exported_symbols_list` or some other mechanism)
+      ##
+
+      ## ----------------------------------------
+      ## BUT ... if that is a problem, try SOLUTION 2, remove the `-bundle-loader` flag
+      ##
+      ## This returns us to the symbol resolution we had in previous Rubies. It feels gross but may
+      ## be a workaround for gem maintainers until we all figure out a better way to deal with this.
+      extdldflags = RbConfig::MAKEFILE_CONFIG["EXTDLDFLAGS"].split
```

Comment:
Wow, what a description! I think that should answer the questions of other RCD users.
Sorry I'm of little help here. Macos is always "kind of Posix" to me, I'm happy if it just works (in CI).

Reply:
:+1: